pkg/migration-tool/planner/slab.go

```diff
@@ -118,6 +144,48 @@ func (s *Slab) Fetch(ctx context.Context, client *utils.Client, mint, maxt int64
 	return nil
 }
 
+func (s *Slab) mergeSubSlabsToSlab(subSlabs []*utils.PrompbResponse) []*prompb.TimeSeries {
+	l := len(subSlabs)
+	s.UpdatePBarMax(s.PBarMax() + 2)
+	s.SetDescription(fmt.Sprintf("combining fetched series from %d responses", l), 1)
+	timeseries := make(map[string]*prompb.TimeSeries)
+	for i := 0; i < l; i++ {
+		ts := subSlabs[i].Result.Timeseries
+		for _, series := range ts {
+			if s, ok := timeseries[getLabelsStr(series.Labels)]; len(series.Samples) > 0 && ok {
+				l := len(s.Samples)
+				if len(s.Samples) > 1 && s.Samples[l-1].Timestamp == series.Samples[0].Timestamp {
+					// Edge case: When concurrent pulls makes the remote-storage send duplicate samples that was sent
```

Comment:
This branch should be an error.

Reply:
Why should it be an error? Do you mean to error and continue with that slab, or error and stop migration?

Reply:
@cevian lets discuss about this.